price-info/src/lib.rs

```diff
@@ -108,50 +108,56 @@ impl<F: Fetch> Client<F> {
 				}
 				Either::B(response.concat2().from_err())
 			})
-			.map(move |body| {
-				let body_str = str::from_utf8(&body).ok();
-				let value: Option<Value> = body_str.and_then(|s| serde_json::from_str(s).ok());
-
-				let ethusd = value
-					.as_ref()
-					.and_then(|value| value.pointer("/result/ethusd"))
-					.and_then(|obj| obj.as_str())
-					.and_then(|s| s.parse().ok());
-
-				match ethusd {
-					Some(ethusd) => {
-						set_price(PriceInfo { ethusd });
-						Ok(())
+			.then(move |body_res| {
+				match body_res {
```

Comment:
This could be replaced with `let body_res = body_res?;` to remove two levels of indentation.